### ISSUE TITLE
Logging in as @wiert@mastodon.social gives error after authorisation

### ISSUE THREAD

**Issue by jpluimers** [diddledani/twitodon#24]
Not sure if this is a bug on the Twitodon side or on mastodon.social`, but I got the below error after these steps:

1. Browse to https://twitodon.com/
2. Authorise Twitter (in my case `@jpluimers`)
3. Authorise Mastodon (in my case `@wiert@mastodon.social`)

The last step then returns to Twitodon as `https://twitodon.com/mastodonAuth?code=<snipped>` whit this error message:

`{"statusCode":500,"error":"Internal Server Error","message":"https://%40wiert@mastodon.social/oauth/token is an url with embedded credentials."}`

Later tried step 3 with `https://mastodon.social/@wiert` which both authorised well, and ended up at https://twitodon.com/ where I could do a successful scan.

--jeroen

**Comment by lucyllewy** [comment]
What did you paste into the mastodon field to authorise in the first instance? Did you type your mastodon account handle (`@wiert@mastodon.social`)?

The field is supposed to take the address of your instance without any usernames attached, so it might be the cause if you did include your handle.

**Comment by jpluimers** [comment]
> What did you paste into the mastodon field to authorise in the first instance? Did you type your mastodon account handle (`@wiert@mastodon.social`)?
> 
> The field is supposed to take the address of your instance without any usernames attached, so it might be the cause if you did include your handle.

Yes, that's what I did. I think in that case the input-field should validate that, as invalidate input and not continue as the continuation results in an incomprehensible error message.

**Comment by lucyllewy** [comment]
Agreed. I'll try to get it sorted.
